### ISSUE TITLE
Cannot install package

### ISSUE THREAD

**Issue by rawinan-soma** [bcbi/ClassImbalance.jl#88]
add ClassImbalance
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package MLJBase [a7f614a8]:
 MLJBase [a7f614a8] log:
 ├─possible versions are: 0.1.0-0.20.2 or uninstalled
 ├─restricted by compatibility requirements with MLJ [add582a8] to versions: [0.1.0-0.2.6, 0.4.0-0.7.5, 0.8.1-0.8.4, 0.9.1-0.12.8, 0.13.1-0.14.9, 0.15.1-0.20.2]
 │ └─MLJ [add582a8] log:
 │   ├─possible versions are: 0.1.0-0.18.2 or uninstalled
 │   ├─restricted to versions * by an explicit requirement, leaving only versions 0.1.0-0.18.2
 │   ├─restricted by compatibility requirements with Tables [bd369af6] to versions: 0.9.2-0.18.2 or uninstalled, leaving only versions: 0.9.2-0.18.2
 │   │ └─Tables [bd369af6] log:
 │   │   ├─possible versions are: 0.1.0-1.7.0 or uninstalled
 │   │   ├─restricted by compatibility requirements with MLJ [add582a8] to versions: 0.1.0-1.7.0
 │   │   │ └─MLJ [add582a8] log: see above
 │   │   └─restricted by compatibility requirements with MLJXGBoostInterface [54119dfa] to versions: 1.0.5-1.7.0
 │   │     └─MLJXGBoostInterface [54119dfa] log:
 │   │       ├─possible versions are: 0.1.1-0.2.0 or uninstalled
 │   │       └─restricted to versions * by an explicit requirement, leaving only versions 0.1.1-0.2.0
 │   └─restricted by compatibility requirements with CategoricalArrays [324d7699] to versions: 0.1.0-0.11.2 or uninstalled, leaving only versions: 0.9.2-0.11.2
 │     └─CategoricalArrays [324d7699] log:
 │       ├─possible versions are: 0.3.11-0.10.5 or uninstalled
 │       ├─restricted by compatibility requirements with MLJ [add582a8] to versions: 0.3.11-0.10.5
 │       │ └─MLJ [add582a8] log: see above
 │       ├─restricted by compatibility requirements with DataFrames [a93c6f00] to versions: 0.3.11-0.7.7
 │       │ └─DataFrames [a93c6f00] log:
 │       │   ├─possible versions are: 0.11.7-1.3.4 or uninstalled
 │       │   ├─restricted to versions * by an explicit requirement, leaving only versions 0.11.7-1.3.4
 │       │   ├─restricted by compatibility requirements with ClassImbalance [04a18a73] to versions: 0.11.7-0.20.2
 │       │   │ └─ClassImbalance [04a18a73] log:
 │       │   │   ├─possible versions are: 0.3.0-0.8.7 or uninstalled
 │       │   │   └─restricted to versions * by an explicit requirement, leaving only versions 0.3.0-0.8.7
 │       │   └─restricted by compatibility requirements with Tables [bd369af6] to versions: [0.11.7-0.13.1, 0.20.1-1.3.4] or uninstalled, leaving only versions: [0.11.7-0.13.1, 0.20.1-0.20.2]
 │       │     └─Tables [bd369af6] log: see above
 │       └─restricted by compatibility requirements with MLJBase [a7f614a8] to versions: 0.7.3-0.10.5, leaving only versions: 0.7.3-0.7.7
 │         └─MLJBase [a7f614a8] log: see above
 ├─restricted by compatibility requirements with Tables [bd369af6] to versions: 0.11.9-0.20.2 or uninstalled, leaving only versions: [0.11.9-0.12.8, 0.13.1-0.14.9, 0.15.1-0.20.2]
 │ └─Tables [bd369af6] log: see above
 ├─restricted by compatibility requirements with CategoricalArrays [324d7699] to versions: 0.7.0-0.13.5 or uninstalled, leaving only versions: [0.11.9-0.12.8, 0.13.1-0.13.5]
 │ └─CategoricalArrays [324d7699] log: see above
 └─restricted by compatibility requirements with MLJModelInterface [e80e1ace] to versions: [0.1.0-0.10.1, 0.13.7-0.20.2] or uninstalled — no versions left
   └─MLJModelInterface [e80e1ace] log:
     ├─possible versions are: 0.1.0-1.4.2 or uninstalled
     └─restricted by compatibility requirements with MLJXGBoostInterface [54119dfa] to versions: 0.3.5-1.4.2
       └─MLJXGBoostInterface [54119dfa] log: see above

**Comment by patrickm663** [comment]
Agree. It seems to be holding locks on out-of-date packages, causing me to have to downgrade or remove conflicting packages (but otherwise up-to-date) in order for it to install.

**Comment by DilumAluthge** [comment]
Yeah. This package needs a lot of work to be updated to work with the latest versions of dependencies. Unfortunately we don't currently have a maintainer for this package.

**Comment by patrickm663** [comment]
I've made a quick [fork](https://github.com/patrickm663/ClassImbalance.jl) because I needed SMOTE for a small ML project I'm working on. This fork recreates the Project.toml from scratch to get the latest versions of this project's packages and switches from `import` to `using`. SMOTE works but I haven't been able to get ROSE to work yet. However, to ever be implemented as a PR the Project.toml would need to be based off of theirs' I figure.

It would need more rigourous testing to make sure the code isn't too deprecated but seemed to run without error for my use-case after relaxing some of the strict typesetting in the function arguments.

**Comment by jerlich** [comment]
Just to add, i have run into a similar problem:
```

(OctagonAgents) pkg> add ClassImbalance
    Updating registry at `~/.julia/registries/General.toml`
   Resolving package versions...
ERROR: Unsatisfiable requirements detected for package DelayEmbeddings [5732040d]:
 DelayEmbeddings [5732040d] log:
 ├─possible versions are: 0.1.0-2.7.4 or uninstalled
 ├─restricted by compatibility requirements with Distributions [31c24e10] to versions: 0.1.0-1.9.0 or uninstalled
 │ └─Distributions [31c24e10] log:
 │   ├─possible versions are: 0.16.0-0.25.103 or uninstalled
 │   ├─restricted to versions * by OctagonAgents [f2f492e4], leaving only versions: 0.16.0-0.25.103
 │   │ └─OctagonAgents [f2f492e4] log:
 │   │   ├─possible versions are: 1.0.0 or uninstalled
 │   │   └─OctagonAgents [f2f492e4] is fixed to version 1.0.0-DEV
 │   └─restricted by compatibility requirements with ClassImbalance [04a18a73] to versions: 0.16.0-0.22.6
 │     └─ClassImbalance [04a18a73] log:
 │       ├─possible versions are: 0.3.0-0.8.7 or uninstalled
 │       └─restricted to versions * by an explicit requirement, leaving only versions: 0.3.0-0.8.7
 ├─restricted by compatibility requirements with StaticArrays [90137ffa] to versions: 1.15.0-2.7.4 or uninstalled, leaving only versions: uninstalled
 │ └─StaticArrays [90137ffa] log:
 │   ├─possible versions are: 0.8.0-1.6.5 or uninstalled
 │   ├─restricted to versions * by OctagonAgents [f2f492e4], leaving only versions: 0.8.0-1.6.5
 │   │ └─OctagonAgents [f2f492e4] log: see above
 │   └─restricted by compatibility requirements with Parquet2 [98572fba] to versions: 1.0.0-1.6.5
 │     └─Parquet2 [98572fba] log:
 │       ├─possible versions are: 0.1.0-0.2.19 or uninstalled
 │       └─restricted to versions * by OctagonAgents [f2f492e4], leaving only versions: 0.1.0-0.2.19
 │         └─OctagonAgents [f2f492e4] log: see above
 └─restricted by compatibility requirements with DynamicalSystems [61744808] to versions: 1.13.0-2.7.4 — no versions left
   └─DynamicalSystems [61744808] log:
     ├─possible versions are: 1.0.0-3.2.3 or uninstalled
     ├─restricted to versions * by OctagonAgents [f2f492e4], leaving only versions: 1.0.0-3.2.3
     │ └─OctagonAgents [f2f492e4] log: see above
     └─restricted by compatibility requirements with StaticArrays [90137ffa] to versions: 1.7.1-3.2.3 or uninstalled, leaving only versions: 1.7.1-3.2.3
       └─StaticArrays [90137ffa] log: see above
```

**Comment by DilumAluthge** [comment]
This package isn't really actively developed any more.

But there is a new package called [Imbalance.jl](https://github.com/JuliaAI/Imbalance.jl) that is being actively developed, so it might be worth checking them out.
